packages/flutter_tools/lib/executable.dart

```diff
@@ -85,7 +85,7 @@ Future<Null> main(List<String> args) async {
     ..addCommand(new UpdatePackagesCommand(hidden: !verboseHelp))
     ..addCommand(new UpgradeCommand());
 
-  return Chain.capture(() async {
+  return Chain.capture/*<Future<Null>>*/(() async {
```

Comment:
Possibly because the callback has no return, and hence we can't infer a return type for it?  That would make the callback typed to return `Future<dynamic>`, which would essentially override the context type of `Future<Null>`.  We've been considering ways to prefer the context type when available, but not sure what we're going to do there yet.